river/sync.go

```diff
@@ -242,13 +242,21 @@ func (r *River) makeUpdateRequest(rule *Rule, rows [][]interface{}) ([]*elastic.
 			req.Action = elastic.ActionDelete
 			reqs = append(reqs, req)
 
-			req = &elastic.BulkRequest{Index: rule.Index, Type: rule.Type, ID: afterID, Parent: afterParentID}
+			req = &elastic.BulkRequest{Index: rule.Index, Type: rule.Type, ID: afterID, Parent: afterParentID, Pipeline: rule.Pipeline}
 			r.makeInsertReqData(req, rule, rows[i+1])
 
 			r.st.DeleteNum.Add(1)
 			r.st.InsertNum.Add(1)
 		} else {
-			r.makeUpdateReqData(req, rule, rows[i], rows[i+1])
+			if len(rule.Pipeline) > 0 {
+				// Pipelines can only be specified on index action
+				r.makeInsertReqData(req, rule, rows[i+1])
```

Comment:
Why use insert here? Do we need to send all columns data here?

Reply:
Yes, Different from update action, index action need all columns data. Otherwise ES will delete non-existent filds